src/test/llmnode_test.cpp

```diff
@@ -335,69 +434,64 @@ TEST_F(LLMFlowHttpTest, unaryChatCompletionsJson) {
     ASSERT_EQ(
         handler->dispatchToProcessor(endpointChatCompletions, requestBody, &response, comp, responseComponents, &writer),
         ovms::StatusCode::OK);
-    rapidjson::Document d;
-    d.Parse(response.c_str());
-    ASSERT_TRUE(d["choices"].IsArray());
-    ASSERT_EQ(d["choices"].Capacity(), 1);
+    parsedResponse.Parse(response.c_str());
+    ASSERT_TRUE(parsedResponse["choices"].IsArray());
+    ASSERT_EQ(parsedResponse["choices"].Capacity(), 1);
     int i = 0;
-    for (auto& choice : d["choices"].GetArray()) {
+    for (auto& choice : parsedResponse["choices"].GetArray()) {
         ASSERT_TRUE(choice["finish_reason"].IsString());
+        EXPECT_STREQ(choice["finish_reason"].GetString(), "length");
         ASSERT_EQ(choice["index"], i++);
         ASSERT_FALSE(choice["logprobs"].IsObject());
         ASSERT_TRUE(choice["message"].IsObject());
         ASSERT_TRUE(choice["message"]["content"].IsString());
-        ASSERT_EQ(choice["message"]["role"], "assistant");
+        EXPECT_STREQ(choice["message"]["role"].GetString(), "assistant");
     }
-    ASSERT_EQ(d["model"], "llmDummyKFS");
-    ASSERT_EQ(d["object"], "chat.completion");
+    EXPECT_STREQ(parsedResponse["model"].GetString(), "llmDummyKFS");
+    EXPECT_STREQ(parsedResponse["object"].GetString(), "chat.completion");
 }
 
-TEST_F(LLMFlowHttpTest, unaryChatCompletionsJsonFinishReasonLength) {
+TEST_F(LLMFlowHttpTest, inferCompletionsStream) {
     std::string requestBody = R"(
         {
             "model": "llmDummyKFS",
-            "stream": false,
+            "stream": true,
             "seed" : 1,
-            "ignore_eos": true,
             "max_tokens": 5,
-            "messages": [
-            {
-                "role": "user",
-                "content": "What is OpenVINO?"
-            }
-            ]
+            "prompt": "What is OpenVINO?"
         }
     )";
-
+    ON_CALL(writer, PartialReply).WillByDefault([this](std::string response) {
+        rapidjson::Document d;
+        ASSERT_EQ(parsePartialReply(response, d), 0);
+        ASSERT_TRUE(d["choices"].IsArray());
+        ASSERT_EQ(d["choices"].Capacity(), 1);
+        int i = 0;
+        for (auto& choice : d["choices"].GetArray()) {
+            if (choice["finish_reason"].IsString()) {
+                EXPECT_STREQ(choice["finish_reason"].GetString(), "length");
```

Comment:
If you want to have this line here, please set `ignore_eos=true` in the sampling params to make it fully deterministic